uwb-beacon-firmware/doc/design_uwb_data_protocol.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Design of a data protocol for CVRA's UWB network
+author: "@antoinealb"
+---
+
+# Problem description
+
+Our robots embed a wireless system based on IEEE 802.15.4 standards.
+Its main purpose was positioning; the wireless modules used (DW1000) allow for high precision time of flight measurements, which can be used in a beacon based position calculation.
+However those modules can also be used for data transmission with the following properties:
+
+* Robust: those modules uses Ultra Wide Band technology, meaning they have more energy and are less susceptible to interference than Wifi.
+    In addition, they can be configured to use other frequencies than traditional radios, making them stronger again.
+* Reasonable bitrate: The manufacturer claims that they can reach 6.8 Mbps.
+    However, this relies on advanced hardware features that our current implementation is not using.
```

Comment:
so in practice we can reach 1Mbps, right?

Reply:
1 Mbps on UAVCAN, so its lower on UWB because of CAN protocol overhead.

Reply:
alright!